app/react/MetadataExtraction/specs/GridChart.spec.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, RenderResult, within } from '@testing-library/react';
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { GridChart } from '../GridChart';
+import * as containerWidthHook from '../useContainerWidthHook';
+
+jest.spyOn(containerWidthHook, 'useContainerWidth').mockReturnValue(50);
+
+const formatForAssertion = (items: HTMLElement[]) => [
```

Comment:
Can you explain this one?

Reply:
Maybe needs refactoring then :D It's just a helper for the test. Because the chart is actually a list of divs, that are draw from left to right (first-up, first-down, second-up, second-down, etc...), this helper turns a list of html elements into a matrix that is easier to work with in the test. For it it just takes the elements in odd positions to be one of the rows, and the ones in even positions to be the other one, and then it just keeps the substring of the text-form of the element that contains the value.